### PR TITLE
changed Time Constant to String; misc code cleanup

### PR DESCRIPTION
**Pull request by kmeyerarendt**
This latest change from Double to String seems counterintuitive but is explicitly suggested in the developer manual for "Updating quantities" and solves the 40 significant digits issue

### CODE REVIEW

**Comment by kmeyerarendt** [comment]
Hi @sandiepeters,
Thanks for your review. 
1. I am unable to reproduce this error. Can you provide the incorrectly formatted Excel file or let me know the steps to create the incorrectly formatted file?
4, 5. I added better error handling and more descriptive error message in case of errors. Let me know whether these are adequate.
6. The two errors I encountered when running the attached Excel file were that Electroporation was incorrectly spelled as Elecroporation and the Competent Cell Id was compound ("00000591, 00000591"). I believe the error messages are now OK (but let me know if you think otherwise) and when I fixed these, the file processed ok

File: .idea/jarRepositories.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sandiepeters) [inline .idea/jarRepositories.xml]: This file probably shouldn't be checked in.

**Comment by kmeyerarendt** [comment]
@sandiepeters I made the error checking and Javadoc changes and removed all the files in the .gitignore file, and committed to development. Please let me know if you can see these changes in this current Pull Request. Thanks!

File: src/main/java/com/genohm/slims/custom/beans/Transformation.java [hunk 2]
```diff
@@ -26,39 +25,105 @@
...
         strVal += optionalColumnName.isEmpty() ? "" : " (" + optionalColumnName + ")";
         return strVal;
     }
 
+    /**
+     * Reads value from either a String or Numeric formatted cell
+     * @param cell   Excel cell
+     * @param columnName        Column header for the input cell
+     * @return String value
+     * @exception   Error out if cell type is neither String nor Numeric
```
Comment (sandiepeters) [inline src/main/java/com/genohm/slims/custom/beans/Transformation.java]: Please indicate the type of exception being thrown.

File: src/main/java/com/genohm/slims/custom/beans/BatchTransformation.java [hunk 3]
```diff
@@ -143,8 +144,9 @@ private SLIMSAccess initializeSlimsAccess() {
...
 
 	/**
-	 * Create a hash table so that Excel column headers can be in any order
-	 * @param row
+	 * Create a hash map so that Excel column headers can be in any order
+	 * @param row	First row of the input Excel file
+	 * @return hash map linking column index and column header name
 	 */
 	public Map<Integer, String>  getColumnHeaders(Row row) {
 		Map<Integer, String> columnHeaders = new HashMap();
```
Comment (sandiepeters) [inline src/main/java/com/genohm/slims/custom/beans/BatchTransformation.java]: Map<Integer, String> columnHeaders = new HashMap() should be Map<Integer, String> columnHeaders = new HashMap<>();

**Comment by kmeyerarendt** [comment]
I have made these changes and committed to the development branch

**Pull request closed without merging**